packages/block-library/src/navigation/use-navigation-menu.js

```diff
@@ -18,9 +18,17 @@ export default function useNavigationMenu( navigationMenuId ) {
 				'wp_navigation',
 				navigationMenuId,
 			];
-			const navigationMenu = navigationMenuId
+
+			let navigationMenu = navigationMenuId
 				? getEditedEntityRecord( ...navigationMenuSingleArgs )
 				: null;
+
+			// getEditedEntityRecord will return the post regardless of status.
+			// Therefore if the found post is not published then we should ignore it.
+			if ( navigationMenuId?.status !== 'publish' ) {
+				navigationMenu = null;
+			}
```

Comment:
I don't think we should allow non-published Navigations to be used. Otherwise you can delete a Navigation and then accidentally have that restored simply by re-opening the Site Editor.